### ISSUE TITLE
Support authentication to GCR/AR

### ISSUE THREAD

**Issue by azhouwd** [runfinch/finch#309]
**What is the problem you're trying to solve?.**
In order to work with [Google Container Registry](https://cloud.google.com/container-registry/) or [Artifact registry](https://cloud.google.com/artifact-registry), Finch needs to be able to authenticate to them.

Google officially provides four authentication ways to Docker:
https://cloud.google.com/artifact-registry/docs/docker/authentication

Nerdctl documents two ways of authenticating to GCR/AR:
https://github.com/containerd/nerdctl/blob/main/docs/registry.md#google-artifact-registry-pkgdev

Currently using the two ways provided from nerdctl do not work for Finch, users need to shell into the VM in order to authenticate to GCR/AR.

When authenticating by using account key in the host OS, the result points out the the credential will be stored in the `/root/.docker/config.json`, which doesn't exist after authenticating and nowhere to access the credential. Changing $DOCKER_CONFIG won't work as well.

When authenticating by auth plugin, it would require to do the steps in the guest OS and doing it in the host OS won't have effect.

**Describe the feature you'd like**
Enable Finch to authenticate to GCR/AR directly in the host operating system.

**Comment by ningziwen** [comment]
Do you know why doesn't something like this work in host? Maybe explain more in the issue for context.
```cat <GCP_SERVICE_ACCOUNT_KEY_JSON> | finch login -u _json_key --password-stdin https://<REGION>-docker.pkg.dev```

**Comment by azhouwd** [comment]
> Do you know why doesn't something like this work in host? Maybe explain more in the issue for context. `cat <GCP_SERVICE_ACCOUNT_KEY_JSON> | finch login -u _json_key --password-stdin https://<REGION>-docker.pkg.dev`

Thanks, I've updated the description with more context.

**Comment by ningziwen** [comment]
My experiment:
In VM, I first tried to `echo $DOCKER_CONFIG`, and it returns empty. That means `/root/.bashrc` is not applied. We will need some setup to apply the env vars in `/root/.bashrc`.
Then I ran
```
export DOCKER_CONFIG="/Users/ningziwe/.finch"
sudo -E nerdctl login registry-1.docker.io
```
The credential is stored in ~/ningziwen/.finch/config.json now. So we may need to add `-E` to limactl command.
-E: https://stackoverflow.com/a/8633575
